src/app/workshops/workshops.component.ts

```diff
@@ -126,28 +180,28 @@ export class WorkshopsComponent {
         if(category != this.categoryList)
         {
 		    this.categoryList = category;
-		    this.updateUrl();
-            console.log(category);
```

Comment:
all the code is repeating for the query params. Extract into a separate function.